src/frontend/src/planner/recursive_union.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::binder::{BoundSetExpr, ShareId};
+use crate::error::Result;
+use crate::optimizer::plan_node::LogicalRecursiveUnion;
+use crate::{PlanRef, Planner};
+
+impl Planner {
+    pub(super) fn plan_recursive_union(
+        &mut self,
+        base: BoundSetExpr,
+        recursive: BoundSetExpr,
+        id: ShareId,
+    ) -> Result<PlanRef> {
+        let base = self.plan_set_expr(base, vec![], &[])?;
+        let recursive = self.plan_set_expr(recursive, vec![], &[])?;
+        let plan = LogicalRecursiveUnion::create(base, recursive);
+        self.ctx.insert_rcte_cache_plan(id, plan.clone());
```

Comment:
I find that the mapping here won't change anymore, so the `PlanRef` always points to a specific plan, considering that the optimizer performs a column pruning optimization on `LogicalRecursiveUnion`, it will generate a different `PlanRef`, but the `rcte_cache` would be stale.

Reply:
then we could update the `rcte_cache` whenever `prune_col` is called for `LogicalRecursiveUnion`.

Reply:
But I didn't see this logic for `LogicalRecursiveUnion` in this PR.

Reply:
It's in #16673 - since I can't update this branch anymore.